datumaro/plugins/camvid_format.py

```diff
@@ -0,0 +1,248 @@
+            label_map[name] = color
+    return label_map
+
+def write_label_map(path, label_map):
+    with open(path, 'w') as f:
+        for label_name, label_desc in label_map.items():
+            f.write('%s %s %s %s\n' % (label_desc[0],
+                label_desc[1], label_desc[2], label_name))
+
+def make_camvid_categories(label_map=None):
+    if label_map is None:
+        label_map = CamvidLabelMap
+    categories = {}
+    label_categories = LabelCategories()
+    for label, desc in label_map.items():
+        label_categories.add(label)
+    categories[AnnotationType.label] = label_categories
+    label_id = lambda label: label_categories.find(label)[0]
+    colormap = { label_id(name): (desc[0], desc[1], desc[2])
+        for name, desc in label_map.items() }
+    mask_categories = MaskCategories(colormap)
+    mask_categories.inverse_colormap # pylint: disable=pointless-statement
+    categories[AnnotationType.mask] = mask_categories
+    return categories
+
+
+class CamvidExtractor(SourceExtractor):
+    def __init__(self, path):
+        assert osp.isfile(path), path
+        self._path = path
+        self._dataset_dir = osp.dirname(path)
+        super().__init__(subset=osp.splitext(osp.basename(path))[0])
+
+
+        self._categories = self._load_categories(self._dataset_dir)
+        self._items = list(self._load_items(path).values())
+
+    @staticmethod
+    def _load_categories(path):
+        label_map = None
+        label_map_path = osp.join(path, CamvidPath.LABELMAP_FILE)
+        if osp.isfile(label_map_path):
+            label_map = parse_label_map(label_map_path)
+        return make_camvid_categories(label_map)
+
+    def _load_items(self, path):
+        items = {}
+        with open(path, encoding='utf-8') as f:
+            for line in f:
+                image, gt = line.split()
+                item_id = ('/'.join(gt.split('/')[2:]))[:-len(CamvidPath.IMAGE_EXT)]
+                image_path = osp.join(self._dataset_dir, image[1:])
+                gt_path = osp.join(self._dataset_dir, gt[1:])
+                inverse_cls_colormap = \
+                    self._categories[AnnotationType.mask].inverse_colormap
+                mask = lazy_mask(gt_path, inverse_cls_colormap)
+                mask = mask()
+                classes = np.unique(mask)
+                item_annotations = []
+                for label_id in classes:
+                    image = self._lazy_extract_mask(mask, label_id)
+                    item_annotations.append(Mask(image=image, label=label_id))
+                items[item_id] = DatasetItem(id=item_id, subset=self._subset,
+                    image=osp.join(image_path), annotations=item_annotations)
+        return items
+
+    @staticmethod
+    def _lazy_extract_mask(mask, c):
+        return lambda: mask == c
+
+
+class CamvidImporter(Importer):
+    @classmethod
+    def find_sources(cls, path):
+        subset_paths = [p for p in glob(osp.join(path, '*.txt'))
```

Comment:
Let's also cover the case with subsets placed somewhere deeper with `**` and `recursive=True`